api/src/main/java/org/apache/iceberg/util/BucketUtil.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.apache.iceberg.relocated.com.google.common.hash.HashFunction;
+import org.apache.iceberg.relocated.com.google.common.hash.Hashing;
+
+/**
+ * Contains the logic for hashing various types for use with the {@code bucket} partition
+ * transformations
+ */
+public class BucketUtil {
+
+  private static final HashFunction MURMUR3 = Hashing.murmur3_32_fixed();
+
+  private BucketUtil() {}
+
+  public static int hashInteger(Integer value) {
```

Comment:
Why do these names include the type? You should be able to just overload `hash`.

Also, why pass values as boxed types here? `HashFunction` accepts primitives, so using a boxed type here means that you'll have to box values passed to `invoke` just to unbox them here.

Reply:
Good catch on both fronts.

For the boxed types, I believe this is a hold over from a previous design when I had the hash function on the Spark subclasses as an abstract function (which didn't work). Will update this evening when I get home :)

Reply:
Updated all to `hash` and changed to use unboxed types.

I might be able to remove some redundant definitions. I'll check.